addons/web/static/src/core/notifications/notification.js

```diff
@@ -31,13 +33,50 @@ export class Notification extends Component {
             },
             optional: true,
         },
+        autocloseDelay: { type: Number, optional: true },
         close: { type: Function },
-        refresh: { type: Function },
-        freeze: { type: Function },
     };
     static defaultProps = {
         buttons: [],
         className: "",
         type: "warning",
+        autocloseDelay: AUTOCLOSE_DELAY,
     };
+    setup() {
+        this.autocloseProgress = useRef("autoclose_progress_bar");
+        this.startNotificationTimer();
+    }
+
+    freeze () {
+        browser.clearTimeout(this.timeout);
+        browser.clearInterval(this.intervalId);
+        this.autocloseProgress.el.style.width = 0;
+    }
+
+    refresh() {
+        this.startNotificationTimer();
+    }
+
+    close() {
+        browser.clearTimeout(this.timeout);
+        browser.clearInterval(this.intervalId);
+        this.props.close();
+    }
+
+    startNotificationTimer() {
+        if (this.props.autocloseDelay === 0) {
+            return
+        }
+
+        this.autocloseDelay = this.props.autocloseDelay;
+        this.intervalId = browser.setInterval(() => {
+            this.autocloseDelay -= 11;
+            if (this.autocloseProgress.el) {
+                this.autocloseProgress.el.style.width = `${(this.autocloseDelay / this.props.autocloseDelay) * 100}%`;
+            }
+        }, 10)
+        this.timeout = browser.setTimeout(() => {
+            this.close();
+        }, this.props.autocloseDelay);
+    }
```

Comment:
this is not reliable. A setInterval/setTimeout doesn't guarantee to take exactly the given delay (e.g. if an heavy processing is ongoing, it may take longer). So 2 suggestions here:
1) use requestAnimationFrame instead of setInterval => will update the DOM only when it is necessary and will ensure not to miss frames
2) to compute the progress in a reliable way, use 3 timestamps:
 - when starting the timer (store it)
 - when the notif should be closed (compute it when starting, and store it): startTime + delay
 - at each animationFrame, compute the ratio using the current timestamp and the two others
 
Also close the notif in the animationFrame, when the current time stamp is over the end timestamp.